sdk/python/kfp/dsl/_container_op.py

```diff
@@ -1071,6 +1075,10 @@ def resolve_artifact_argument(artarg):
         # decorator func to proxy a method in `Container` into `ContainerOp`
         def _proxy(proxy_attr):
             """Decorator func to proxy to ContainerOp.container"""
+            # Stop proxy for some methods in v2.
```

Comment:
Can you help me understand the main goal of this change?

Perhaps in the future the IR container spec will be getting more and more capabilities of Kubernetes' ContainerSpec and PodSpec, so maybe following that structure has some benefits.

Reply:
I also think it might be easier to just enable overriding these methods in the derived classes.

Just add `If not hasattr(self, attr_to_proxy):` on line 1096. Then the methods that already exist won't be replaced with proxies.

Reply:
> I also think it might be easier to just enable overriding these methods in the derived classes.

Exactly. What I observed is after `_proxy` the methods in vs.ContainerOp is redirected to the one in `dsl.ContainerOp`. Let me try your suggestion.

Reply:
Done.